src/Chapter1.hs

```diff
@@ -542,7 +541,11 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Ord a => a -> a -> a -> a
+mid x y z
+     | min y z <= x && x <= max y z = x
+     | min x z <= y && y <= max x z = y
+     | min x y <= z && z <= max x y = z
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = z`, to tell the compiler, that your pattern matching is exhaustive 🙂


It looks like you can move all these 3 cases under a single otherwise and return z in here.
The implementation will be a bit shorter 🙂